src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/CQLSAIIndex.java

```diff
@@ -128,29 +135,68 @@ static IndexTarget indexTarget(IndexMetadata indexMetadata)
 
-    var target = CQLSAIIndex.Options.TARGET.getWithDefault(indexMetadata.getOptions());
-    Matcher matcher = INDEX_TARGET_PATTERN.matcher(target);
-    if (!matcher.matches()) {
-      throw new UnsupportedCqlIndexException(
-          "Could not parse index target: '" + target + "'", indexMetadata);
-    }
-
+    // if the regex matches then the target is in the form "keys(foo)", "entries(bar)" etc
+    // if not, then it must be a simple column name and implicitly its type is VALUES
+    var target = indexMetadata.getTarget();
+    Matcher matcher = TARGET_REGEX.matcher(target);
     String columnName;
-    String functionName;
-    if (matcher.group(2) == null) {
-      columnName = matcher.group(1);
-      functionName = null;
-    } else {
+    CqlIndexType cqlIndexType;
+    if (matcher.matches()) {
+      cqlIndexType = CqlIndexType.fromString(matcher.group(1));
       columnName = matcher.group(2);
-      functionName = matcher.group(1);
+    } else {
+      columnName = target;
+      cqlIndexType = CqlIndexType.VALUES;
     }
 
-    return functionName == null
-        ? new IndexTarget(CqlIdentifier.fromInternal(columnName), null)
-        : new IndexTarget(
-            CqlIdentifier.fromInternal(columnName), ApiIndexFunction.fromCql(functionName));
+    // 1. In the case of a quoted column name the name in the target string
+    // will be enclosed in quotes, which we need to unwrap.
+    // 2. It may also include quote characters internally, escaped like so:
+    // abc"def -> abc""def, then we need to un-escape as abc"def -> abc"def
+    // to get the actual column name
+    if (columnName.startsWith(QUOTE)) {
+      columnName = StringUtils.substring(StringUtils.substring(columnName, 1), 0, -1);
+      columnName = TWO_QUOTES.matcher(columnName).replaceAll(QUOTE);
+    }
+
+    return new IndexTarget(CqlIdentifier.fromInternal(columnName), cqlIndexType);
   }
 
   /** For internal to this package use only */
-  protected record IndexTarget(CqlIdentifier targetColumn, ApiIndexFunction indexFunction) {}
+  public record IndexTarget(CqlIdentifier targetColumn, CqlIndexType cqlIndexType) {}
+
+  public enum CqlIndexType {
+    VALUES,
+    KEYS,
+    KEYS_AND_VALUES,
+    FULL,
+    SIMPLE;
+
+    public String toString() {
+      switch (this) {
+        case KEYS:
+          return "keys";
+        case KEYS_AND_VALUES:
+          return "entries";
+        case FULL:
+          return "full";
+        case VALUES:
+          return "values";
+        case SIMPLE:
+          return "";
+        default:
+          return "";
+      }
+    }
+
+    public static CqlIndexType fromString(String s) {
+      if ("".equals(s)) return SIMPLE;
+      else if ("values".equals(s)) return VALUES;
+      else if ("keys".equals(s)) return KEYS;
+      else if ("entries".equals(s)) return KEYS_AND_VALUES;
+      else if ("full".equals(s)) return FULL;
+
+      throw new AssertionError("Unrecognized index target type " + s);
```

Comment:
This should be IllegalArgument we have not used AssertionError anywhere in the code